test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -749,6 +755,19 @@ private static Streamable tryCreateNewInstance(Streamable streamable) throws NoS
         }
     }
 
+    private static Streamable tryCreateFromStream(Streamable streamable, StreamInput in) throws NoSuchMethodException,
+        InstantiationException, IllegalAccessException, InvocationTargetException {
```

Comment:
Can you indent this one more time so it doesn't look like method body?

Reply:
Actually, I don't think think you throw these exceptions any more since you catch and return null.

Reply:
I think you might want to let InvocationTargetException bubble up instead of return null. Those are probably serialization errors.

Reply:
Or, better, maybe only return null if there isn't a StreamInput ctor, otherwise you should use it.

I think when we remove the `Streamable` interface we'd want to remove the null return entirely and require such a constructor here.

Reply:
I've updated as Ryan suggested. so now all those are thrown again. And yes, once all are migrated. the original `tryCreateNewInstance` will be removed